presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueues.java

```diff
@@ -239,6 +241,34 @@ private static void cancelQuery(DistributedQueryRunner queryRunner, QueryId quer
         queryRunner.getCoordinator().getQueryManager().cancelQuery(queryId);
     }
 
+    private static void waitForCompleteQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, TERMINAL_QUERY_STATES, expectedCount);
+    }
+
+    private static void waitForRunningQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, ImmutableSet.of(RUNNING), expectedCount);
+    }
+
+    private static void waitForQueryCount(DistributedQueryRunner queryRunner, Set<QueryState> countingStates, int expectedCount)
+            throws InterruptedException
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        int count;
+        do {
+            MILLISECONDS.sleep(500);
+            count = 0;
+            for (QueryInfo queryInfo : queryManager.getAllQueryInfo()) {
+                if (countingStates.contains(queryInfo.getState())) {
+                    count += 1;
+                }
+            }
+        } while (count != expectedCount);
```

Comment:
The test framework will kill the timeout test and throw a ``ThreadTimeoutException``. I think one thing we can do is to print more useful error message on timeout. However, that should be a separate issue since such kinds of wait-timeout patterns are used in other test cases. We may want to change them all at once.